lamindb_setup/core/upath.py

```diff
@@ -548,3 +548,15 @@ def get_stat_dir_gs(path: UPath) -> Tuple[int, str, str, int]:
     n_objects = len(md5s)
     hash, hash_type = hash_md5s_from_dir(md5s)
     return sum(sizes), hash, hash_type, n_objects
+
+
+def check_s3_storage_location_empty(path: UPath) -> bool:
+    objects = path.fs.find(path.as_posix(), detail=True)
+    n_objects = len(objects.values())
```

Comment:
Is this the fastest way to get the number of objects? Can you add a comment re alternatives?

Reply:
I can't guarantee with 100% certainty here - I just know it's faster than leveraging a `boto3` session via `s3.Bucket(bucket).objects.filter(Prefix=key)` (based on our benchmark: https://lamin.ai/laminlabs/lamindata/transform/krGp3hT1f78N5zKv).

@Koncopd any additional thoughts here?

Reply:
I just pushed a simple update that uses the same method with `detail=False`. I'm not sure if it improves performance, but I would expect it to.